spec/e2e/id_property_spec.rb

```diff
@@ -340,38 +366,153 @@ def foobar
     end
   end
 
+  describe 'id_property :neo_id' do
+    before do
+      before_session do
+        stub_const('NeoIdTest', UniqueClass.create do
+          include Neo4j::ActiveNode
+          id_property :neo_id
+        end)
+      end
+    end
+
+    it 'has an index' do
+      expect(NeoIdTest.mapped_label.indexes[:property_keys]).to be_empty
+    end
+
+    describe 'property id' do
+      it 'is not defined when before save ' do
+        node = NeoIdTest.new
+        expect(node.id).to be_nil
+      end
+
+      it 'is is set when saving ' do
+        node = NeoIdTest.new
+        node.save
+        expect(node.id).to be_present
```

Comment:
This is clear if you do something like

```ruby
expect { node.save }.to change { node.id.present? }.from(true).to(false)
```

It'd also let you remove the example above that checks for `nil` prior to save.